File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/masterdata/infrastructure/jpa/inmemory/MasterDataRepositoryInMemory.java
package io.zenwave360.example.clinicaltool.modules.masterdata.infrastructure.jpa.inmemory;

import io.zenwave360.example.clinicaltool.modules.masterdata.core.domain.*;
import io.zenwave360.example.clinicaltool.modules.masterdata.core.outbound.jpa.MasterDataRepository;
import java.math.*;
import java.time.*;
import java.util.*;

public class MasterDataRepositoryInMemory extends InMemoryJpaRepository<MasterData> implements MasterDataRepository {
    @Override
    public java.util.Optional<MasterData> findByTypeAndKey(MasterDataType type, String key) {
        return getEntities().values().stream().filter(e ->
             isSameValue(type, readField(e, "type")) && isSameValue(key, readField(e, "key")) 
        ).findFirst();
    }
}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/zenwave360/example/clinicaltool/modules/masterdata/inmemory/MasterDataRepositoryInMemory.kt
package io.zenwave360.example.clinicaltool.modules.masterdata.inmemory

import io.zenwave360.example.clinicaltool.modules.masterdata.MasterDataRepository
import io.zenwave360.example.clinicaltool.modules.masterdata.domain.*
import java.math.*
import java.time.*
import java.util.*

class MasterDataRepositoryInMemory : InMemoryJpaRepository<MasterData, Long>(), MasterDataRepository {
    override fun findByTypeAndKey(type: MasterDataType, key: String): MasterData? {
        return getEntities().values.firstOrNull { e ->
            isSameValue(type, readField(e, "type")) && isSameValue(key, readField(e, "key"))
        }
    }
}
